packages/strapi-plugin-content-manager/admin/src/containers/EditView/index.js

```diff
@@ -1,7 +1,714 @@
+      const filesMap = getMediaAttributes(layout, groupLayoutsData);
+      // Create the formdata to upload all the files
+      const formDatas = Object.keys(filesMap).reduce((acc, current) => {
+        const keys = current.split('.');
+        const isMultiple = get(filesMap, [current, 'multiple'], false);
+        const isGroup = get(filesMap, [current, 'isGroup'], false);
+        const isRepeatable = get(filesMap, [current, 'repeatable'], false);
+
+        const getFilesToUpload = path => {
+          const value = get(modifiedData, path, []) || [];
+
+          return value.filter(file => {
+            return file instanceof File;
+          });
+        };
+        const getFileToUpload = path => {
+          const file = get(modifiedData, [...path, 0], '');
+          if (file instanceof File) {
+            return [file];
+          }
+
+          return [];
+        };
+
+        if (!isRepeatable) {
+          const currentFilesToUpload = isMultiple
+            ? getFilesToUpload(keys)
+            : getFileToUpload([...keys]);
+
+          if (!isEmpty(currentFilesToUpload)) {
+            acc[current] = currentFilesToUpload.reduce((acc2, curr) => {
+              acc2.append('files', curr);
+
+              return acc2;
+            }, new FormData());
+          }
+        }
+
+        if (isGroup && isRepeatable) {
+          const [key, targetKey] = current.split('.');
+          const groupData = get(modifiedData, [key], []);
+          const groupFiles = groupData.reduce((acc1, current, index) => {
+            const files = isMultiple
+              ? getFileToUpload([key, index, targetKey])
+              : getFileToUpload([key, index, targetKey]);
+
+            if (!isEmpty(files)) {
+              const toFormData = files.reduce((acc2, curr) => {
+                acc2.append('files', curr);
+
+                return acc2;
+              }, new FormData());
+
+              acc1[`${key}.${index}.${targetKey}`] = toFormData;
+            }
+
+            return acc1;
+          }, {});
+
+          return { ...acc, ...groupFiles };
+        }
+
+        return acc;
+      }, {});
+      // Change the request helper default headers so we can pass a FormData
+      const headers = { 'X-Forwarded-Host': 'strapi' };
+      // Upload the files
+      const mapUploadedFiles = Object.keys(formDatas).reduce(
+        async (acc, current) => {
+          const collection = await acc;
+          try {
+            const uploadedFiles = await request(
+              '/upload',
+              {
+                method: 'POST',
+                body: formDatas[current],
+                headers,
+                signal: submitSignal,
+              },
+              false,
+              false
+            );
+
+            collection[current] = uploadedFiles;
+
+            return collection;
+          } catch (err) {
+            console.log('upload error', err);
```

Comment:
I left the logs in purpose since we are not handling the errors from the back-end correctly right now. However, if you think that they should be removed I'll do it